### PR TITLE
Pipeline Automation: create, run, delete

### PR DESCRIPTION
**Pull request by slickwarren**
ran flake8, added environment variables

added clarifying comments

moved test def to top of definitions, added delete test case
moved test to correct place, moved pipeline delete to main test

### CODE REVIEW

**Review by izaac — CHANGES_REQUESTED** [review]
Looks good besides a couple comments.

File: tests/validation/tests/v3_api/test_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    pipeline_details["pipeline_run"] = pipeline_details["p_client"].action(obj=pipeline,
+                                                             action_name='run',
+                                                             branch="master")
+    wait_for_condition(
+        pipeline_details["p_client"], pipeline, check_last_run_state("Success"),
+        timeout=PIPELINE_TIMEOUT)
+    assert len(pipeline_view_logs()) > 1
+    print("Cleaning up...")
+    pipeline_details["p_client"].delete(pipeline)
+    assert len(pipeline_details["p_client"].list_pipeline(
```
[the remark this answers is not in the document]
Comment (bmdepesa) [reply tests/validation/tests/v3_api/test_pipeline.py:30]: We want the delete to be explicitly called as part of the actual test here, not in the finalizer.

[the remark this answers is not in the document]
Comment (izaac) [reply tests/validation/tests/v3_api/test_pipeline.py:30]: Shouldn't we then have separate tests for Create and Delete?

[the remark this answers is not in the document]
Comment (bmdepesa) [reply tests/validation/tests/v3_api/test_pipeline.py:30]: The only difference then between the two cases would be the removal of the log assertion in the `delete` case, which seems redundant.

[the remark this answers is not in the document]
Comment (izaac) [reply tests/validation/tests/v3_api/test_pipeline.py:30]: We followed that pattern on backups that's why I brought it up here.
https://github.com/rancher/rancher/blob/master/tests/validation/tests/v3_api/test_bkp_restore_local.py

**Comment by slickwarren** [comment]
closing in favor of #31633

File: tests/validation/tests/v3_api/test_pipeline.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (slickwarren) [reply tests/validation/tests/v3_api/test_pipeline.py:40]: Is this resolved, or do I need to update something, @izaac

[the remark this answers is not in the document]
Comment (izaac) [reply tests/validation/tests/v3_api/test_pipeline.py:40]: keep it as-is for now, we can revisit later @slickwarren

**Review by izaac — APPROVED** [review]

**Review by sowmyav27 — APPROVED** [review]
LGTM

**Review by bmdepesa — APPROVED** [review]
LGTM

**Pull request merged**